app/models/manageiq/providers/amazon/cloud_manager/refresher.rb

```diff
@@ -26,6 +26,14 @@ def parse_targeted_inventory(ems, _target, inventory)
     hashes, = Benchmark.realtime_block(:parse_inventory) do
       if refresher_options.try(:[], :inventory_object_refresh)
         ManageIQ::Providers::Amazon::CloudManager::RefreshParserInventoryObject.new(inventory).populate_inventory_collections
+
+        if inventory.kind_of?(ManageIQ::Providers::Amazon::Inventory::Targets::EmsEventCollection)
+          # For EmsEventCollection, we want to parse everything, regardless whether it belongs to Cloud or Network
+          # Manager
+          ManageIQ::Providers::Amazon::NetworkManager::RefreshParserInventoryObject.new(inventory).populate_inventory_collections
```

Comment:
Ah, yeah, sorry, I answered to something else it seems. Yes. In here, it should parse all data, if this was an event based refresh.